src/coreclr/jit/valuenum.cpp

```diff
@@ -11130,8 +11130,11 @@ void Compiler::fgValueNumberTree(GenTree* tree)
                     if (isExact && (handle != NO_CLASS_HANDLE))
                     {
                         JITDUMP("IND(obj) is actually a class handle for %s\n", eeGetClassName(handle));
-                        // Filter out all shared generic instantiations
-                        if ((info.compCompHnd->getClassAttribs(handle) & CORINFO_FLG_SHAREDINST) == 0)
+                        uint32_t attrs = info.compCompHnd->getClassAttribs(handle);
+
+                        // Filter out all shared generic instantiations and non-exact arrays
```

Comment:
What is the difference between `isExact` that we have checked above and `impIsClassExact` that we are repeating here?

Reply:
`isExact` means that JIT knows the exact type of the tree (e.g. it sees that it's created via `new`). `impIsClassExact` is supposed to mean that a class is sealed (or "effectively sealed")